### PR TITLE
WIP: Add MSI Support for Azure HostedClusters

### PR DESCRIPTION
**Pull request by bryan-cox**
**What this PR does / why we need it**:
This PR adds:
* API support for the client IDs related to the MSIs for cluster-image-registry, cluster-ingress, cluster-storage,
 and cluster-network operators
* Adds an environment variable AZURE_MSI_AUTHENTICATION, set to true when the client ID is provided, in the deployments of cluster-image-registry, cluster-ingress, cluster-storage,
 and cluster-network operators

**Which issue(s) this PR fixes**:
Fixes #

**Checklist**
- [x] Subject and description added to both, commit and PR.
- [x] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

File: api/hypershift/v1alpha1/hostedcluster_types.go [hunk 1]
```diff
@@ -1574,6 +1574,46 @@ type AzurePlatformSpec struct {
...
+}
+
+type ControlPlaneManagedServiceIdentities struct {
+	// ImageRegistryMSIClientID is the client ID of a pre-existing managed identity ID of that will be associated with
+	// the cluster-image-registry-operator. The managed identity will be in a different resource group other than
+	// ResourceGroupName.
+	//
+	// +kubebuilder:validation:Required
+	// +required
+	ImageRegistryMSIClientID string `json:"imageRegistryMSIClientID,omitempty"`
```
Comment (muraee) [inline api/hypershift/v1alpha1/hostedcluster_types.go:1592]: required fields should not have `omitempty`
same comment for the rest of the fields.

**Comment by bryan-cox** [comment]
/close
Working on this through https://github.com/openshift/hypershift/pull/4690

**Pull request closed without merging**